File: example/src/main/java/eu/darken/mvpbakery/example/screens/MainModule.java
package eu.darken.mvpbakery.example.screens;

import androidx.fragment.app.Fragment;
import dagger.Module;
import dagger.Provides;
import eu.darken.mvpbakery.example.screens.counting.CountingFragment;

@Module
public class MainModule {

    @Provides
    @MainComponent.Scope
    Class<? extends Fragment> provideStartFragment() {
        return CountingFragment.class;
    }
}

File: example/src/main/java/eu/darken/mvpbakery/example/screens/MainModule.kt
package eu.darken.mvpbakery.example.screens

import androidx.fragment.app.Fragment
import dagger.Module
import dagger.Provides
import eu.darken.mvpbakery.example.screens.counting.CountingFragment

@Module
class MainModule {

    @Provides
    @MainComponent.Scope
    internal fun provideStartFragment(): Class<out Fragment> {
        return CountingFragment::class.java
    }
}
